From 655f9dce3630828e43c7893d59e82de0ff2b2386 Mon Sep 17 00:00:00 2001
From: Jai Luthra <j-luthra@ti.com>
Date: Fri, 14 Jun 2024 15:19:46 +0530
Subject: [PATCH 061/222] media: ti: j721e-csi2rx: Fix missing v4l2 cleanup

If probe fails after registering the v4l2 device and subdev, we need to
clean it up. This got accidentally removed while separating out the
driver to support multiple contexts.

Fixes: 3f8fc7379649 ("media: ti: j721e-csi2rx: separate out device and context")
Signed-off-by: Jai Luthra <j-luthra@ti.com>
---
 drivers/media/platform/ti/j721e-csi2rx/j721e-csi2rx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/media/platform/ti/j721e-csi2rx/j721e-csi2rx.c b/drivers/media/platform/ti/j721e-csi2rx/j721e-csi2rx.c
index cea0360a9d42..8a135f9645c4 100644
--- a/drivers/media/platform/ti/j721e-csi2rx/j721e-csi2rx.c
+++ b/drivers/media/platform/ti/j721e-csi2rx/j721e-csi2rx.c
@@ -1613,7 +1613,7 @@ static int ti_csi2rx_probe(struct platform_device *pdev)
 	i--;
 	for (; i >= 0; i--)
 		ti_csi2rx_cleanup_ctx(&csi->ctx[i]);
-
+	ti_csi2rx_cleanup_v4l2(csi);
 err_v4l2:
 	mutex_destroy(&csi->mutex);
 	dma_free_coherent(csi->dev, csi->drain.len, csi->drain.vaddr,
-- 
2.45.2

